content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +35,643 @@ are covered here as well as the `System.Diagnostics` API.
-    Activity.Current = null;
+    public ActivitySource activitySource;
+    private int min;
+    private int max;
 
-    var newRoot = MyActivitySource.StartActivity("NewRoot");
+    public Dice(int min, int max, ActivitySource activitySource)
+    {
+        this.min = min;
+        this.max = max;
+        this.activitySource = activitySource;
+    }
 
-    // Re-set the previous Current Activity so the trace isn't messed up
-    Activity.Current = previous;
+    //...
 }
 ```
 
-### Get the current Activity
+### Create Activities
 
-Sometimes it's helpful to access whatever the current `Activity` is at a point
-in time so you can enrich it with more information.
+Now that you have [activitySources](/docs/concepts/signals/traces/#tracer) initialized, you can create [activities](/docs/concepts/signals/traces/#spans).
 
-```csharp
-var activity = Activity.Current;
-// may be null if there is none
-```
+The code below illustrates how to create an activity.
 
-Note that `using` is not used in the prior example. Doing so will end current
-`Activity`, which is not likely to be desired.
+```csharp
+public List<int> rollTheDice(int rolls)
+{   
+    List<int> results = new List<int>();
+    
+    // It is recommended to create activities, only when doing operations that are worth measuring independently. 
+    // Too many activities makes it harder to visualize in tools like Jaeger.
+    using (var myActivity = activitySource.StartActivity("rollTheDice"))
+    {
+        for (int i = 0; i < rolls; i++)
+        {
+            results.Add(rollOnce());
+        }
 
-### Add tags to an Activity
+        return results;
+    }
+}
 
-Tags (the equivalent of
-[`Attributes`](/docs/concepts/signals/traces/#attributes) in OpenTelemetry) let
-you attach key/value pairs to an `Activity` so it carries more information about
-the current operation that it's tracking.
+```
+If you followed the instructions using the [example app](#example-app) up to
+this point, you can copy the code above in your library file `Dice.cs`. You
+should now be able to see activities/spans emitted from your app.
 
-```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
+Start your app as follows, and then send it requests by visiting <http://localhost:8080/rolldice?rolls=12> with your browser or curl.
 
-activity?.SetTag("operation.value", 1);
-activity?.SetTag("operation.name", "Saying hello!");
-activity?.SetTag("operation.other-stuff", new int[] { 1, 2, 3 });
+```sh
+dotnet run
 ```
 
-We recommend that all Tag names are defined in constants rather than defined
-inline as this provides both consistency and also discoverability.
+After a while, you should see the spans printed in the console by the
+`ConsoleExporter`, something like this:
+
+```json
+Activity.TraceId:            841d70616c883db82b4ae4e11c728636
+Activity.SpanId:             9edfe4d69b0d6d8b
+Activity.TraceFlags:         Recorded
+Activity.ParentSpanId:       39fcd105cf958377
```

Comment:
given this is the child Activity, would it better to show the parent activity also here.